lib/ansible/modules/cloud/amazon/elasticache_parameter_group.py

```diff
@@ -0,0 +1,336 @@
+            "description": "initial description"
+        },
+        "response_metadata": {
+            "http_headers": {
+                "content-length": "562",
+                "content-type": "text/xml",
+                "date": "Mon, 06 Feb 2017 22:14:08 GMT",
+                "x-amzn-requestid": "947291f9-ecb9-11e6-85bd-3baa4eca2cc1"
+            },
+            "http_status_code": 200,
+            "request_id": "947291f9-ecb9-11e6-85bd-3baa4eca2cc1",
+            "retry_attempts": 0
+        }
+changed:
+    description: if the cache parameter group has changed
+    returned: always
+    type: bool
+    sample:
+        "changed": true
+"""
+
+
+from ansible.module_utils.ec2 import boto3_conn
+
+try:
+    import boto3
+    import botocore
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+def create(module, conn, name, group_family, description):
+    """ Create ElastiCache parameter group. """
+    try:
+        response = conn.create_cache_parameter_group(CacheParameterGroupName=name, CacheParameterGroupFamily=group_family, Description=description)
+        changed = True
+    except boto.exception.ClientError as e:
+        module.fail_json(msg=e.message)
+    return response, changed
+
+def delete(module, conn, name):
+    """ Delete ElastiCache parameter group. """
+    try:
+        conn.delete_cache_parameter_group(CacheParameterGroupName=name)
+        response = {}
+        changed = True
+    except boto.exception.ClientError as e:
+        module.fail_json(msg=e.message)
+    return response, changed
+
+def make_current_modifiable_param_dict(conn, name):
+    """ Gets the current state of the cache parameter group and creates a dict with the format: {ParameterName: [Allowed_Values, DataType, ParameterValue]}"""
+    current_info = get_info(conn, name)
+    if current_info is False:
+        return False
+
+    parameters = current_info["Parameters"]
+    modifiable_params = {}
+
+    for param in parameters:
+        if param["IsModifiable"] and ("AllowedValues" and "ParameterValue") in param:
+            modifiable_params[param["ParameterName"]] = [param["AllowedValues"], param["DataType"], param["ParameterValue"]]
+    return modifiable_params
+
+def check_valid_modification(module, values, modifiable_params):
+    """ Check if the parameters and values in values are valid.  """
+    changed_with_update = False
+
+    for param_value_pair in values:
+        parameter = param_value_pair[0]
+        new_value = param_value_pair[1]
+
+        # check valid modifiable parameters
+        if parameter not in modifiable_params.keys():
+            module.fail_json("%s is not a modifiable parameter. Valid parameters to modify are: %s." % (parameter, modifiable_params.keys()))
+
+        # check allowed datatype for modified parameters
+        str_to_type = {"integer": int, "string": str}
+        if type(new_value) != str_to_type[modifiable_params[parameter][1]]:
```

Comment:
Instead of type checks like this, use the `isinstance` operator like:

```
if isinstance(new_value, str_to_type[modifiable_params[parameter][1]]):`
```